mmf/datasets/processors/text_image_processors.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+from math import nan
+
+import numpy as np
+import torch
+from mmf.common.registry import registry
+from mmf.datasets.processors.processors import BaseProcessor
+from mmf.utils.process_text_image import text_token_overlap_with_bbox
+from mmf.utils.text import is_punctuation
+from scipy import interpolate
+from transformers.tokenization_auto import AutoTokenizer
+
+
+class TextImageProcessor(BaseProcessor):
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(config, *args, **kwargs)
+        self.config = config
+        text_tokenizer_config = config.text_processor.tokenizer_config
+        self.text_tokenizer = AutoTokenizer.from_pretrained(
+            text_tokenizer_config.type, **text_tokenizer_config.params
+        )
+
+    def sample_traces(self, utterances, utterance_times, traces):
+        tokens, index_tokens, token_weights, token_traces = [], [], [], []
+        for idx, utterance in enumerate(utterances):
+            if self.config.remove_punctuations:
+                utterance = self.remove_punctuation(utterance)
```

Comment:
I think this will be slow as we check every char if it's punctuation or not. Here are some alternative solutions that might work and should be faster : https://stackoverflow.com/questions/265960/best-way-to-strip-punctuation-from-a-string

`utterance.translate(None, string.punctuation)`